src/couch/src/couch_server.erl

```diff
@@ -860,19 +865,55 @@ get_engine_path(DbName, Engine) when is_binary(DbName), is_atom(Engine) ->
     end.
 
 lock(DbName, Reason) when is_binary(DbName), is_binary(Reason) ->
-    case ets:lookup(couch_dbs, DbName) of
+    case ets:lookup(couch_dbs(DbName), DbName) of
         [] ->
-            true = ets:insert(couch_dbs_locks, {DbName, Reason}),
+            true = ets:insert(couch_dbs_locks(DbName), {DbName, Reason}),
             ok;
         [#entry{}] ->
             {error, already_opened}
     end.
 
 unlock(DbName) when is_binary(DbName) ->
-    true = ets:delete(couch_dbs_locks, DbName),
+    true = ets:delete(couch_dbs_locks(DbName), DbName),
     ok.
 
 
+db_updated(Db) ->
+    DbName = couch_db:name(Db),
+    gen_server:call(couch_server(DbName), {db_updated, Db}, infinity).
+
+
+couch_server(Arg) ->
+    name("couch_server", Arg).
+
+
+couch_dbs(Arg) ->
+    name("couch_dbs", Arg).
+
+
+couch_dbs_pid_to_name(Arg) ->
+    name("couch_dbs_pid_to_name", Arg).
+
+
+couch_dbs_locks(Arg) ->
+    name("couch_dbs_locks", Arg).
+
+
+name(BaseName, DbName) when is_binary(DbName) ->
+    N = 1 + (erlang:crc32(DbName) rem num_servers()),
```

Comment:
```1 + erlang:phash2(DbName,  num_servers())``` might be cleaner here since we are not really computing a checksum but want to do hashing

Reply:
May even be a tiny bit faster...

```
> DbName = crypto:strong_rand_bytes(100).
<<23,241,236,53,92,151,12,217,246,88,65,153,224,214,236,
  240,126,243,188,248,53,187,206,165,235,96,24,5,38,...>>

> f(USec), {USec, ok} = timer:tc(fun() -> [erlang:crc32(DbName) rem 10 || I <- lists:seq(1, 1000000)], ok end), USec/1.0e6.
1.499477

> f(USec), {USec, ok} = timer:tc(fun() -> [erlang:phash2(DbName, 10) || I <- lists:seq(1, 1000000)], ok end), USec/1.0e6.
1.331179
```

Reply:
hm, that looks nice, sure. I don't think checksum vs hash is a strong argument though, they're both means to an end.

Reply:
@nickva I'm having trouble finding a reference, but I've been under the impression that `timer:tc` is not to be trusted on anonymous functions in the shell and should be tested on a compiled module.

Reply:
@chewbranca I think they'd both be hit by the same slowness...

I tried a module mostly out of curiosity:
```erlang
-module(test_hash).

-export([test_crc32/2, test_phash/2]).

test_crc32(N, Val) ->
    T0 = erlang:system_time(),
    do_crc32(N, Val),
    usec_per_cycle(erlang:system_time() - T0, N).

test_phash(N, Val) ->
    T0 = erlang:system_time(),
    do_phash2(N, Val),
    usec_per_cycle(erlang:system_time() - T0, N).

do_crc32(0, _) ->
    ok;
do_crc32(N, Val) ->
    _ = erlang:crc32(Val) rem 10,
    do_crc32(N - 1, Val).

do_phash2(0, _) ->
    ok;
do_phash2(N, Val) ->
    _ = erlang:phash2(Val, 10),
    do_phash2(N - 1, Val).

usec_per_cycle(Native, N) ->
    erlang:convert_time_unit(Native, native, microsecond) / N.
```

```
18> test_hash:test_crc32(10000000, DbName).
0.0722587
19> test_hash:test_phash(10000000, DbName).
0.070593
```